provider/runbook_data_test.go

```diff
@@ -0,0 +1,102 @@
+package provider
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccRunbookDataSource_basic(t *testing.T) {
+	rName := acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testFireHydrantIsSetup(t) },
+		ProviderFactories: defaultProviderFactories(),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccRunbookDataSourceConfig_basic(rName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttrSet("data.firehydrant_runbook.test_runbook", "id"),
+					resource.TestCheckResourceAttr(
+						"data.firehydrant_runbook.test_runbook", "name", fmt.Sprintf("test-runbook-%s", rName)),
+				),
+			},
+		},
+	})
+}
+
+func TestAccRunbookDataSource_allAttributes(t *testing.T) {
```

Comment:
Would this be the proper name? Seems like this is all top level attributes but doesn't include testing the steps. Maybe because this is just attributes owned by the runbook model it self?

Reply:
Ideally it would eventually be all attributes, including the ones in the steps, but after looking at the data source, it currently only supports id, name, and description. It just throws away the information about steps. If we add support for the steps to the data source, we should def update this test to include them.